File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/AuthorizedDataSourcesMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.model.AuthorizedDataSources
import com.rookmotion.rook_sdk_health_connect.proto.AuthorizedDataSourcesProto

fun AuthorizedDataSources.toProto(): AuthorizedDataSourcesProto {
    return AuthorizedDataSourcesProto.newBuilder()
        .setOura(oura.toInt())
        .setPolar(polar.toInt())
        .setWhoop(whoop.toInt())
        .setFitbit(fitbit.toInt())
        .setGarmin(garmin.toInt())
        .setWithings(withings.toInt())
        .setDexcom(dexcom.toInt())
        .setAppleHealth(appleHealth.toInt())
        .setHealthConnect(healthConnect.toInt())
        .setAndroid(android.toInt())
        .build()
}

File: packages/rook_sdk_health_connect/lib/src/data/mapper/authorized_data_sources_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/extension/int_extensions.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pb.dart';

extension AuthorizedDataSourceMappers on AuthorizedDataSourcesProto {
  AuthorizedDataSources toDomain() {
    return AuthorizedDataSources(
      oura: oura.toBoolOrNull(),
      polar: polar.toBoolOrNull(),
      whoop: whoop.toBoolOrNull(),
      fitbit: fitbit.toBoolOrNull(),
      garmin: garmin.toBoolOrNull(),
      withings: withings.toBoolOrNull(),
      dexcom: dexcom.toBoolOrNull(),
      appleHealth: appleHealth.toBoolOrNull(),
      healthConnect: healthConnect.toBoolOrNull(),
      android: android.toBoolOrNull(),
    );
  }
}
